src/diffusers/pipelines/stable_diffusion/safety_checker.py

```diff
@@ -65,16 +69,20 @@ def forward(self, clip_input, images):
 
             result.append(result_img)
 
-        has_nsfw_concepts = [len(res["bad_concepts"]) > 0 for res in result]
+        has_nsfw_concepts = torch.tensor([len(res["bad_concepts"]) > 0 for res in result])
 
         for idx, has_nsfw_concept in enumerate(has_nsfw_concepts):
             if has_nsfw_concept:
-                images[idx] = np.zeros(images[idx].shape)  # black image
+                images[idx] = torch.zeros(images[idx].shape)  # black image
 
         if any(has_nsfw_concepts):
             logger.warning(
                 "Potential NSFW content was detected in one or more images. A black image will be returned instead."
                 " Try again with a different prompt and/or seed."
             )
 
+        if is_input_numpy:
```

Comment:
```suggestion
        if isinstance(images, np.ndarray):
```

Reply:
this is no longer `True` here (see the comment above)

Reply:
Ahh I understand! Could we maybe change the naming to `if return_format == "np"`?  and above to `return_format = "np" if isintance(..., np.ndarray) else "pt"`

Think this could be a bit more understandable for me at least, but it's a nit. Ok to leave for me. Think return format is a bit nicer as one reading the code at this point here cares more about what should be returned and not what was put in